lib/cursor/ChangeStream.js

```diff
@@ -54,8 +54,8 @@ class ChangeStream extends EventEmitter {
 
         ['close', 'change', 'end', 'error'].forEach(ev => {
           this.driverChangeStream.on(ev, data => {
-            if (data.fullDocument != null && this.options && this.options.hydrate) {
-              data.fullDocument = this.options.model.hydrate(data.fullDocument);
+            if (this.options && this.options.fullDocument != null && this.options.hydrate) {
+              data.fullDocument = this.options.model.hydrate(this.options.fullDocument);
```

Comment:
This is very wrong. The whole point here is to hydrate `data.fullDocument`, if it exists. `this.options.fullDocument` is a boolean, not the document to hydrate.

Reply:
Ah, my bad. I misinterpreted the intent code 😅 
Just fixed it! Still working the same.
Thanks!